app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,70 @@
+require 'date'
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    # task_updates = params[:task]
+    # @task = Task.find(params[:id])
+    # @task.update_attributes(task_updates)
+    # @task.save
+
+    @task = Task.find(params[:id])
+    @task.name = params[:task][:name]
+    @task.description = params[:task][:description]
+    @task.completion_date = params[:task][:completion_date]
+    task_params
+    @task.save
+
+    redirect_to task_path(@task)
+    # redirect_to('/tasks')
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(
+    name: params[:task][:name], 
+    description: params[:task][:description], completion_date: params[:task][:completion_date])
+    task.save
+    redirect_to('/tasks')
+  end
+
+  def destroy
+    @task = Task.find(params[:id])
+    @task.destroy
+    redirect_to('/tasks')
```

Comment:
Ideally here we would use `redirect_to tasks_path` instead. It should work the same way as this line does, but is more flexible if we for some reason need to change the URL that maps to the `tasks#index` action.